### PR TITLE
Add pagination for murren all

### PR DESCRIPTION
**Pull request by Murrengan**

### CODE REVIEW

**Comment by ydreiz** [comment]
Pagination does not work on the last page.

**My suggestions for making changes to the logic**

**_src/views/Home.vue_**
```
    data: () => ({
      loading: true,
      showSignUpMurrens: [],
      nextPageNumber: 1,
      nextPageExists: null,
    }),
    async mounted() {
      await this.fetchMurren()
      this.loading = false
    },
    methods: {
      async fetchMurren(nextPageNumber = null) {
        const {next, results} = await this.$store.dispatch('fetchMurrens', nextPageNumber)

        this.showSignUpMurrens = [...this.showSignUpMurrens, ...results]
        this.nextPageExists = next
      },
      async murrDefOnIntersected() {
        if (this.nextPageExists) {
          await this.fetchMurren(this.nextPageNumber)
        }
      }
    },
    watch: {
      nextPageExists(exists) {
        if (exists) {
          this.nextPageNumber++;
        }
      }
    },
```
**_src/store/index.js_**
```
    async fetchMurrens(_, page = null) {
      try {
        let url = '/murren/all/'

        if (page) {
          url += `?page=${page}`
        }

        const {data} = await axios.get(url)
        return data
      } catch (error) {
        return {error: true, message: 'Ошибка на сервере'}
      }
    },
```

This approach monitors the `next` variable from the backend and if it does not equal `null` then pagination starts working

File: src/views/Home.vue [hunk 1]
```diff
@@ -30,16 +32,35 @@
...
-      this.signUpMurrens = result ? result.reverse() : []
+      const result = await this.$store.dispatch('fetchMurrens', this.pageNumber)
+      this.showSignUpMurrens = result ? result.results : []
+      this.allSignUpMurrens = Math.ceil(result.count / result.results.length)
+      this.pageNumber++
       this.loading = false
     },
+    methods: {
+      async murrDefOnIntersected() {
+        if (this.pageNumber < this.allSignUpMurrens) {
```
Comment (ydreiz) [inline src/views/Home.vue]: One more page remains, the condition needs to be corrected from `<` to `<=`

**Review by ydreiz — CHANGES_REQUESTED** [review]

File: src/views/Home.vue (hunk 1 above)
Comment (Murrengan) [reply src/views/Home.vue]: done

**Comment by Murrengan** [comment]
@DemonOFmooN Great idea - implemented

File: src/components/common/Observer.vue [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    }),
+    mounted() {
+      this.observer = new IntersectionObserver(([entry]) => {
+        if (entry && entry.isIntersecting) {
+          this.$emit('murrIntersect')
+        }
+      })
+
+      this.observer.observe(this.$el)
+    }
```
Comment (ydreiz) [inline src/components/common/Observer.vue:21]: I think you need to add disconnection
```
    destroyed() {
      this.observer.disconnect()
    }
```

**Review by ydreiz — CHANGES_REQUESTED** [review]

File: src/components/common/Observer.vue (hunk 2 above)
Comment (Murrengan) [reply src/components/common/Observer.vue:21]: done

**Pull request merged**